server/server.go

```diff
@@ -74,47 +79,109 @@ func (a *ArgoCDServer) Run() {
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
+	grpcS := a.newGRPCServer()
+	httpS := a.newHTTPServer(ctx)
```

Comment:
Will we expose the http to outside since https seems to be always there?

Reply:
We should not serve any content when running in HTTP mode. I still need to implement a 302 redirect handler if the connection muxes to HTTP.